### PR TITLE
create getLogs wrapper

### PR DESCRIPTION
**Pull request by tota79**

### CODE REVIEW

**Comment by leovigna** [comment]
Discussed a bit with @OscBacon 
In the indexer actions for ERC20/721/1155  it better to:
- Not define fromBlock / toBlock => This is unspecified behaviour depending on node configuration?
- Use latest block height as toBlock and fromBlock as that number minus 10k

File: packages/backend-indexer/src/actions/indexer/getLogs.ts [hunk 1]
```diff
@@ -13,19 +14,21 @@ export async function getLogs<
     client: Client<Transport, chain>,
     params: GetLogsParameters<abiEvent, abiEvents, strict, fromBlock, toBlock>,
 ): Promise<GetLogsReturnType<abiEvent, abiEvents, strict, fromBlock, toBlock>> {
-    const { fromBlock, toBlock } = params;
+    const toBlock = params.toBlock ?? BigInt(await client.request({ method: "eth_blockNumber" }));
```
Comment (leovigna) [inline packages/backend-indexer/src/actions/indexer/getLogs.ts]: Update to using getBlockNumber action from viem

File: packages/backend-indexer/src/actions/indexer/getLogs.ts [hunk 2]
```diff
@@ -13,19 +14,21 @@ export async function getLogs<
...
+
+    let fromBlock = params.fromBlock ?? (typeof toBlock === "bigint" ? toBlock - 10_000n : 0n);
```
Comment (leovigna) [inline packages/backend-indexer/src/actions/indexer/getLogs.ts]: Default value for fromBlock should be undefined no?

**Review by leovigna — APPROVED** [review]
lgtm

File: packages/backend-indexer/src/actions/indexer/erc1155.ts [hunk 3]
```diff
@@ -29,16 +30,15 @@ export async function getERC1155Tokens<chain extends Chain | undefined>(
         throw new Error("One of address or account must be specified");
     }
 
-    const getLogsAction = getAction(client, getLogs, "getLogs");
+    // const getLogsAction = getAction(client, getLogs, "getLogs");
```
Comment (OscBacon) [inline packages/backend-indexer/src/actions/indexer/erc1155.ts]: remove commented out code

File: packages/backend-indexer/src/actions/indexer/erc20.ts [hunk 4]
```diff
@@ -28,8 +29,13 @@ export async function getERC20Tokens<chain extends Chain | undefined>(
         throw new Error("One of address or account must be specified");
     }
 
-    const getLogsAction = getAction(client, getLogs, "getLogs");
-    const logs = await getLogsAction({ address, event: Transfer, args: { to: account }, strict: true, fromBlock: 0n });
+    // const getLogsAction = getAction(client, getLogs, "getLogs");
```
Comment (OscBacon) [inline packages/backend-indexer/src/actions/indexer/erc20.ts]: remove commented out code

File: packages/backend-indexer/src/actions/indexer/erc721.ts [hunk 5]
```diff
@@ -30,8 +31,13 @@ export async function getERC721Tokens<chain extends Chain | undefined>(
         throw new Error("One of address or account must be specified");
     }
 
-    const getLogsAction = getAction(client, getLogs, "getLogs");
-    const logs = await getLogsAction({ address, event: Transfer, args: { to: account }, strict: true, fromBlock: 0n });
+    // const getLogsAction = getAction(client, getLogs, "getLogs");
```
Comment (OscBacon) [inline packages/backend-indexer/src/actions/indexer/erc721.ts]: remove commented out code

File: packages/backend-indexer/src/actions/indexer/getLogs.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    const toBlock = params.toBlock ?? (await getBlockNumber(client));
+
+    let fromBlock = params.fromBlock ?? (typeof toBlock === "bigint" ? toBlock - 10_000n : undefined);
+
+    if (typeof fromBlock === "bigint") {
+        fromBlock = fromBlock < 0n ? 0n : fromBlock;
+    }
+
+    if (typeof fromBlock === "bigint" && typeof toBlock === "bigint" && toBlock - fromBlock > 10_000n) {
+        throw new Error("Block range too large. Ensure `toBlock - fromBlock < 10,000`.");
```
Comment (OscBacon) [inline packages/backend-indexer/src/actions/indexer/getLogs.ts]: -> "Block range must be less than 10,000 blocks"

**Review by OscBacon — CHANGES_REQUESTED** [review]

File: packages/backend-indexer/src/actions/indexer/getLogs.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        ? [abiEvent]
+        : undefined,
+    strict extends boolean | undefined = undefined,
+    fromBlock extends BlockNumber | BlockTag | undefined = undefined,
+    toBlock extends BlockNumber | BlockTag | undefined = undefined,
+>(
+    client: Client<Transport, chain>,
+    params: GetLogsParameters<abiEvent, abiEvents, strict, fromBlock, toBlock>,
+): Promise<GetLogsReturnType<abiEvent, abiEvents, strict, fromBlock, toBlock>> {
+    const toBlock = params.toBlock ?? (await getBlockNumber(client));
```
Comment (OscBacon) [inline packages/backend-indexer/src/actions/indexer/getLogs.ts]: Default for toBlock is latest. Shouldn't you get the block number if `toBlock` is `latest` for consistency?

**Review by OscBacon — CHANGES_REQUESTED** [review]

File: packages/backend-indexer/src/actions/indexer/getLogs.ts (hunk 7 above)
Comment (leovigna) [reply packages/backend-indexer/src/actions/indexer/getLogs.ts]: Fair point

File: packages/backend-indexer/src/actions/indexer/getLogs.ts [hunk 8]
```diff
@@ -14,21 +14,26 @@ export async function getLogs<
     client: Client<Transport, chain>,
     params: GetLogsParameters<abiEvent, abiEvents, strict, fromBlock, toBlock>,
 ): Promise<GetLogsReturnType<abiEvent, abiEvents, strict, fromBlock, toBlock>> {
-    const toBlock = params.toBlock ?? (await getBlockNumber(client));
+    let resolvedToBlock: BlockNumber | BlockTag | undefined;
```
Comment (OscBacon) [inline packages/backend-indexer/src/actions/indexer/getLogs.ts]: When is resolvedToBlock undefined?

**Review by OscBacon — CHANGES_REQUESTED** [review]

**Review by OscBacon — APPROVED** [review]

**Pull request merged**